src/docmanager/index.ts

```diff
@@ -219,11 +219,11 @@ interface IModelFactory {
  * The document manager.
  */
 export
-class DocumentManager {
+class DocumentManager<T extends Widget> {
```

Comment:
Yep, what I really should have done was make the types in the functions `IWidgetFactory<Widget>`. I fixed it and pushed again.